DNN test_cumsum "conformance" test fails in 5.x branch

### System Information

OpenCV 5.x Debug build
Platform: any


### Detailed description

The issue was triggered by adding 0-d and 1-d support to cv::Mat in 

The assertion is triggered in https://github.com/opencv/opencv/blob/d789cb459ca7fdfd02bbbe89b8f164d4e46b3a21/modules/dnn/src/layer_internals.hpp#L167. There is no dimension `size[1]`

### Steps to reproduce

1. Remove related tests from `modules/dnn/test/test_onnx_conformance_layer_parser_denylist.inl.hpp`
2. Build and test:
```
cmake -DCMAKE_BUILD_TYPE=Debug ../opencv-5.x
make -j4 opencv_test_dnn
./opencv_test_dnn --gtest_filter=test_cumsum*
```

Output example:
```
[  FAILED  ] 21 tests, listed below:
[  FAILED  ] Test_ONNX_conformance.Layer_Test/test_cumsum_1d_OCV_OCL, where GetParam() = (test_cumsum_1d, OCV/OCL)
[  FAILED  ] Test_ONNX_conformance.Layer_Test/test_cumsum_1d_OCV_OCL_FP16, where GetParam() = (test_cumsum_1d, OCV/OCL_FP16)
[  FAILED  ] Test_ONNX_conformance.Layer_Test/test_cumsum_1d_OCV_CPU, where GetParam() = (test_cumsum_1d, OCV/CPU)
[  FAILED  ] Test_ONNX_conformance.Layer_Test/test_cumsum_1d_exclusive_OCV_OCL, where GetParam() = (test_cumsum_1d_exclusive, OCV/OCL)
[  FAILED  ] Test_ONNX_conformance.Layer_Test/test_cumsum_1d_exclusive_OCV_OCL_FP16, where GetParam() = (test_cumsum_1d_exclusive, OCV/OCL_FP16)
[  FAILED  ] Test_ONNX_conformance.Layer_Test/test_cumsum_1d_exclusive_OCV_CPU, where GetParam() = (test_cumsum_1d_exclusive, OCV/CPU)
[  FAILED  ] Test_ONNX_conformance.Layer_Test/test_cumsum_1d_reverse_OCV_OCL, where GetParam() = (test_cumsum_1d_reverse, OCV/OCL)
[  FAILED  ] Test_ONNX_conformance.Layer_Test/test_cumsum_1d_reverse_OCV_OCL_FP16, where GetParam() = (test_cumsum_1d_reverse, OCV/OCL_FP16)
[  FAILED  ] Test_ONNX_conformance.Layer_Test/test_cumsum_1d_reverse_OCV_CPU, where GetParam() = (test_cumsum_1d_reverse, OCV/CPU)
[  FAILED  ] Test_ONNX_conformance.Layer_Test/test_cumsum_1d_reverse_exclusive_OCV_OCL, where GetParam() = (test_cumsum_1d_reverse_exclusive, OCV/OCL)
[  FAILED  ] Test_ONNX_conformance.Layer_Test/test_cumsum_1d_reverse_exclusive_OCV_OCL_FP16, where GetParam() = (test_cumsum_1d_reverse_exclusive, OCV/OCL_FP16)
[  FAILED  ] Test_ONNX_conformance.Layer_Test/test_cumsum_1d_reverse_exclusive_OCV_CPU, where GetParam() = (test_cumsum_1d_reverse_exclusive, OCV/CPU)
[  FAILED  ] Test_ONNX_conformance.Layer_Test/test_cumsum_2d_axis_0_OCV_OCL, where GetParam() = (test_cumsum_2d_axis_0, OCV/OCL)
[  FAILED  ] Test_ONNX_conformance.Layer_Test/test_cumsum_2d_axis_0_OCV_OCL_FP16, where GetParam() = (test_cumsum_2d_axis_0, OCV/OCL_FP16)
[  FAILED  ] Test_ONNX_conformance.Layer_Test/test_cumsum_2d_axis_0_OCV_CPU, where GetParam() = (test_cumsum_2d_axis_0, OCV/CPU)
[  FAILED  ] Test_ONNX_conformance.Layer_Test/test_cumsum_2d_axis_1_OCV_OCL, where GetParam() = (test_cumsum_2d_axis_1, OCV/OCL)
[  FAILED  ] Test_ONNX_conformance.Layer_Test/test_cumsum_2d_axis_1_OCV_OCL_FP16, where GetParam() = (test_cumsum_2d_axis_1, OCV/OCL_FP16)
[  FAILED  ] Test_ONNX_conformance.Layer_Test/test_cumsum_2d_axis_1_OCV_CPU, where GetParam() = (test_cumsum_2d_axis_1, OCV/CPU)
[  FAILED  ] Test_ONNX_conformance.Layer_Test/test_cumsum_2d_negative_axis_OCV_OCL, where GetParam() = (test_cumsum_2d_negative_axis, OCV/OCL)
[  FAILED  ] Test_ONNX_conformance.Layer_Test/test_cumsum_2d_negative_axis_OCV_OCL_FP16, where GetParam() = (test_cumsum_2d_negative_axis, OCV/OCL_FP16)
[  FAILED  ] Test_ONNX_conformance.Layer_Test/test_cumsum_2d_negative_axis_OCV_CPU, where GetParam() = (test_cumsum_2d_negative_axis, OCV/CPU)
```

Details:
```
[==========] Running 1 test from 1 test case.
[----------] Global test environment set-up.
[----------] 1 test from Test_ONNX_conformance
[ RUN      ] Test_ONNX_conformance.Layer_Test/test_cumsum_1d_OCV_CPU, where GetParam() = (test_cumsum_1d, OCV/CPU)
[ INFO:0@0.248] global onnx_importer.cpp:803 populateNet DNN/ONNX: loading ONNX v7 model produced by 'backend-test'. Number of nodes = 1, initializers = 0, inputs = 2, outputs = 1
[ INFO:0@0.248] global onnx_importer.cpp:696 parseOperatorSet DNN/ONNX: ONNX opset version = 13
[ INFO:0@0.248] global onnx_importer.cpp:974 handleNode DNN/ONNX: processing node with 2 inputs and 1 outputs: [CumSum]:(onnx_node_output_0!y) from domain='ai.onnx'
Exception during net.forward() call!
unknown file: Failure
C++ exception with description "OpenCV(5.0.0-pre) /home/alexander/Projects/OpenCV/opencv-next/modules/core/include/opencv2/core/mat.inl.hpp:1294: error: (-215:Assertion failed) i < dims() in function 'operator[]'
" thrown in the test body.
[  FAILED  ] Test_ONNX_conformance.Layer_Test/test_cumsum_1d_OCV_CPU, where GetParam() = (test_cumsum_1d, OCV/CPU) (2 ms)
[----------] 1 test from Test_ONNX_conformance (3 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test case ran. (4 ms total)
[  PASSED  ] 0 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] Test_ONNX_conformance.Layer_Test/test_cumsum_1d_OCV_CPU, where GetParam() = (test_cumsum_1d, OCV/CPU)
```

Details:
```
[==========] Running 1 test from 1 test case.
[----------] Global test environment set-up.
[----------] 1 test from Test_ONNX_conformance
[ RUN      ] Test_ONNX_conformance.Layer_Test/test_cumsum_1d_OCV_CPU, where GetParam() = (test_cumsum_1d, OCV/CPU)
[ INFO:0@0.248] global onnx_importer.cpp:803 populateNet DNN/ONNX: loading ONNX v7 model produced by 'backend-test'. Number of nodes = 1, initializers = 0, inputs = 2, outputs = 1
[ INFO:0@0.248] global onnx_importer.cpp:696 parseOperatorSet DNN/ONNX: ONNX opset version = 13
[ INFO:0@0.248] global onnx_importer.cpp:974 handleNode DNN/ONNX: processing node with 2 inputs and 1 outputs: [CumSum]:(onnx_node_output_0!y) from domain='ai.onnx'
Exception during net.forward() call!
unknown file: Failure
C++ exception with description "OpenCV(5.0.0-pre) /home/alexander/Projects/OpenCV/opencv-next/modules/core/include/opencv2/core/mat.inl.hpp:1294: error: (-215:Assertion failed) i < dims() in function 'operator[]'
" thrown in the test body.
[  FAILED  ] Test_ONNX_conformance.Layer_Test/test_cumsum_1d_OCV_CPU, where GetParam() = (test_cumsum_1d, OCV/CPU) (2 ms)
[----------] 1 test from Test_ONNX_conformance (3 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test case ran. (4 ms total)
[  PASSED  ] 0 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] Test_ONNX_conformance.Layer_Test/test_cumsum_1d_OCV_CPU, where GetParam() = (test_cumsum_1d, OCV/CPU)
```

### Issue submission checklist

- [X] I report the issue, it's not a question
- [X] I checked the problem with documentation, FAQ, open issues, forum.opencv.org, Stack Overflow, etc and have not found any solution
- [X] I updated to the latest OpenCV version and the issue is still there
- [X] There is reproducer code and related data files (videos, images, onnx, etc)

--- a/modules/dnn/src/layers/cumsum_layer.cpp
+++ b/modules/dnn/src/layers/cumsum_layer.cpp
@@ -36,7 +36,7 @@ public:
                          std::vector<MatShape> &outputs,
                          std::vector<MatShape> &internals) const CV_OVERRIDE
     {
-        Layer::getMemoryShapes(inputs, requiredOutputs, outputs, internals);
+        outputs.assign(1, inputs[0]);
         return exclusive_raw == 0;
     }
 

--- a/modules/dnn/test/test_onnx_conformance_layer_parser_denylist.inl.hpp
+++ b/modules/dnn/test/test_onnx_conformance_layer_parser_denylist.inl.hpp
@@ -438,11 +438,3 @@
 "test_unsqueeze_three_axes",  // ---- same as above ---
 "test_unsqueeze_two_axes",   // ---- same as above ---)
 "test_unsqueeze_unsorted_axes",  // ---- same as above ---)
-// // Cumsum related issue: https://github.com/opencv/opencv/issues/24437
-"test_cumsum_1d", //Issue:: output shape creation mismatch
-"test_cumsum_1d_exclusive",  // ---- same as above ---
-"test_cumsum_1d_reverse",  // ---- same as above ---
-"test_cumsum_1d_reverse_exclusive",  // ---- same as above ---
-"test_cumsum_2d_axis_0",  // ---- same as above ---
-"test_cumsum_2d_axis_1",  // ---- same as above ---
-"test_cumsum_2d_negative_axis",  // ---- same as above ---

